flink-clients/src/main/java/org/apache/flink/client/deployment/application/ApplicationDispatcherBootstrap.java

```diff
@@ -159,6 +160,17 @@ public void stop() {
                         })
                 .exceptionally(
                         t -> {
+                            final Optional<JobStartupFailedException> jobStartupFailedOpt =
```

Comment:
Is this the right commit for the change? Maybe it should be 3 commits. I more thought that the scope of this commit is ending with PackageProgram...

Reply:
Hmm, the change is not that big anymore. Do you see it as a blocker?